ovos_utils/geolocation.py

```
@@ -0,0 +1,249 @@
+        Dict[str, Any]: JSON structure with lookup results.
+
+    Raises:
+        ConnectionError: If the geolocation service cannot be reached.
+        ValueError: If the service returns empty results.
+    """
+    url = "https://nominatim.openstreetmap.org/search"
+    try:
+        response = requests.get(url, params={"q": location, "format": "json", "limit": 1},
+                                headers={"User-Agent": "OVOS/1.0"}, timeout=timeout)
+    except (RequestException, Timeout) as e:
+        raise ConnectionError(f"Failed to connect to geolocation service: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)